src/custom/pages/Profile/LockedGnoVesting/hooks.ts

```diff
@@ -44,17 +33,22 @@ export const useAllocation = (): CurrencyAmount<Token> => {
           setAllocation(CurrencyAmount.fromRawAmount(COW, claim?.amount ?? 0))
         }
       })
+    } else {
+      setAllocation(initial)
     }
     return () => {
       canceled = true
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
which dep is it bitching about?

Reply:
`initial` but I cant add it because its object and it will be created in each render and trigger an infinite re-renders

Reply:
ok. can make it a ref also

Reply:
is it solved if you use useRef?

Reply:
echo echo echo

Reply:
Yep ref worked :+1: